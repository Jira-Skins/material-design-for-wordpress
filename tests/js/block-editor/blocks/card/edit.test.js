/**
 * External dependencies
 */
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';

/**
 * Internal dependencies
 */
import Edit from '../../../../../assets/src/block-editor/blocks/card/edit';

/**
 * Render the component.
 *
 * @param {Object} props - Component props.
 * @return {Function} A functional component.
 */
const setup = props => {
	return render( <Edit { ...props } /> );
};

const baseProps = {
	attributes: {
		cardLayout: 'vertical',
		contentLayout: 'text-under-media',
		title: 'Test title',
		displayTitle: true,
		secondaryText: 'Test secondary text',
		displaySecondaryText: true,
		imageSourceUrl: 'http://test.loc/test.jpg',
		isImageEditMode: false,
		displayImage: true,
		supportingText: 'Test supporting text',
		displaySupportingText: true,
		primaryActionButtonLabel: 'Test button action 1',
		primaryActionButtonUrl: 'http://test.loc/hello-world',
		primaryActionButtonNewTab: true,
		primaryActionButtonNoFollow: true,
		secondaryActionButtonLabel: 'Test button action 2',
		secondaryActionButtonUrl: 'http://test.loc/hello-world-again',
		secondaryActionButtonNewTab: true,
		secondaryActionButtonNoFollow: true,
		displayActions: true,
		displaySecondaryActionButton: true,
		outlined: true,
		cornerRadius: 4,
	},
};

describe( 'blocks: material/card: Edit', () => {
	it( 'matches snapshot', () => {
		const wrapper = setup( baseProps );
		expect( wrapper ).toMatchSnapshot();
	} );

	it( 'matches snapshot with a `horizontal` card layout.', () => {
		const props = { ...baseProps };
		baseProps.attributes.cardLayout = 'horizontal';
		const wrapper = setup( props );
		expect( wrapper ).toMatchSnapshot();
	} );

	it( 'matches snapshot when props get defaults', () => {
		const props = { ...baseProps };
		delete props.attributes.cardLayout;
		const wrapper = setup( props );
		expect( wrapper ).toMatchSnapshot();
	} );
} );
