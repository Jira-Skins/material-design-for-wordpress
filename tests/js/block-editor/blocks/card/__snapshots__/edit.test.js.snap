// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`blocks: material/card: Edit matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      aria-atomic="true"
      aria-live="assertive"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-assertive"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div
      aria-atomic="true"
      aria-live="polite"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-polite"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div>
      <div>
        <div
          class="mdc-card mtb-card mdc-card--outlined"
          style="border-radius: 4px;"
        >
          <div
            class="mdc-card__primary-action mtb-card__primary-action mdc-ripple-upgraded"
            tabindex="0"
          >
            <div
              class="mtb-card__media-container"
            >
              <div
                class="mdc-card__media mdc-card__media--16-9 mtb-card__media mtb-card-with-text-under-media"
                style="background-image: url(http://test.loc/test.jpg);"
                tabindex="0"
              />
              <div
                class="mtb-card__media-actions"
              >
                <button
                  aria-label="Remove Card Image"
                  class="components-button mtb-card__media-close-button has-icon"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="dashicon dashicons-no"
                    focusable="false"
                    height="20"
                    role="img"
                    viewBox="0 0 20 20"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M12.12 10l3.53 3.53-2.12 2.12L10 12.12l-3.54 3.54-2.12-2.12L7.88 10 4.34 6.46l2.12-2.12L10 7.88l3.54-3.53 2.12 2.12z"
                    />
                  </svg>
                </button>
              </div>
            </div>
            <div
              class="mtb-card__primary"
            >
              <h2
                aria-label="Title goes here"
                aria-multiline="true"
                class="rich-text block-editor-rich-text__editable mtb-card__title mdc-typography mdc-typography--headline6"
                contenteditable="true"
                role="textbox"
                style="white-space: pre-wrap;"
              >
                Test title
              </h2>
              <h3
                aria-label="Secondary text"
                aria-multiline="true"
                class="rich-text block-editor-rich-text__editable mtb-card__secondary-text mdc-typography mdc-typography--subtitle2"
                contenteditable="true"
                role="textbox"
                style="white-space: pre-wrap;"
              >
                Test secondary text
              </h3>
            </div>
            <div
              aria-label="Supporting text"
              aria-multiline="true"
              class="rich-text block-editor-rich-text__editable mtb-card__supporting-text mtb-card__supporting-text_text-under-media mdc-typography mdc-typography--body2"
              contenteditable="true"
              role="textbox"
              style="white-space: pre-wrap;"
            >
              Test supporting text
            </div>
          </div>
          <div
            class="mdc-card__actions"
          >
            <div
              class="mdc-card__action-buttons"
            >
              <div
                class="mdc-card__action-button-container"
              >
                <button
                  class="mdc-button mdc-card__action mdc-card__action--button"
                >
                  <div
                    aria-label="Button text"
                    aria-multiline="true"
                    class="rich-text block-editor-rich-text__editable"
                    contenteditable="true"
                    role="textbox"
                    style="white-space: pre-wrap;"
                  >
                    Test button action 1
                  </div>
                </button>
              </div>
              <div
                class="mdc-card__action-button-container"
              >
                <button
                  class="mdc-button mdc-card__action mdc-card__action--button"
                >
                  <div
                    aria-label="Button text"
                    aria-multiline="true"
                    class="rich-text block-editor-rich-text__editable"
                    contenteditable="true"
                    role="textbox"
                    style="white-space: pre-wrap;"
                  >
                    Test button action 2
                  </div>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="mdc-card mtb-card mdc-card--outlined"
        style="border-radius: 4px;"
      >
        <div
          class="mdc-card__primary-action mtb-card__primary-action mdc-ripple-upgraded"
          tabindex="0"
        >
          <div
            class="mtb-card__media-container"
          >
            <div
              class="mdc-card__media mdc-card__media--16-9 mtb-card__media mtb-card-with-text-under-media"
              style="background-image: url(http://test.loc/test.jpg);"
              tabindex="0"
            />
            <div
              class="mtb-card__media-actions"
            >
              <button
                aria-label="Remove Card Image"
                class="components-button mtb-card__media-close-button has-icon"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="dashicon dashicons-no"
                  focusable="false"
                  height="20"
                  role="img"
                  viewBox="0 0 20 20"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12.12 10l3.53 3.53-2.12 2.12L10 12.12l-3.54 3.54-2.12-2.12L7.88 10 4.34 6.46l2.12-2.12L10 7.88l3.54-3.53 2.12 2.12z"
                  />
                </svg>
              </button>
            </div>
          </div>
          <div
            class="mtb-card__primary"
          >
            <h2
              aria-label="Title goes here"
              aria-multiline="true"
              class="rich-text block-editor-rich-text__editable mtb-card__title mdc-typography mdc-typography--headline6"
              contenteditable="true"
              role="textbox"
              style="white-space: pre-wrap;"
            >
              Test title
            </h2>
            <h3
              aria-label="Secondary text"
              aria-multiline="true"
              class="rich-text block-editor-rich-text__editable mtb-card__secondary-text mdc-typography mdc-typography--subtitle2"
              contenteditable="true"
              role="textbox"
              style="white-space: pre-wrap;"
            >
              Test secondary text
            </h3>
          </div>
          <div
            aria-label="Supporting text"
            aria-multiline="true"
            class="rich-text block-editor-rich-text__editable mtb-card__supporting-text mtb-card__supporting-text_text-under-media mdc-typography mdc-typography--body2"
            contenteditable="true"
            role="textbox"
            style="white-space: pre-wrap;"
          >
            Test supporting text
          </div>
        </div>
        <div
          class="mdc-card__actions"
        >
          <div
            class="mdc-card__action-buttons"
          >
            <div
              class="mdc-card__action-button-container"
            >
              <button
                class="mdc-button mdc-card__action mdc-card__action--button"
              >
                <div
                  aria-label="Button text"
                  aria-multiline="true"
                  class="rich-text block-editor-rich-text__editable"
                  contenteditable="true"
                  role="textbox"
                  style="white-space: pre-wrap;"
                >
                  Test button action 1
                </div>
              </button>
            </div>
            <div
              class="mdc-card__action-button-container"
            >
              <button
                class="mdc-button mdc-card__action mdc-card__action--button"
              >
                <div
                  aria-label="Button text"
                  aria-multiline="true"
                  class="rich-text block-editor-rich-text__editable"
                  contenteditable="true"
                  role="textbox"
                  style="white-space: pre-wrap;"
                >
                  Test button action 2
                </div>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`blocks: material/card: Edit matches snapshot when props get defaults 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      aria-atomic="true"
      aria-live="assertive"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-assertive"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div
      aria-atomic="true"
      aria-live="polite"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-polite"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div>
      <div>
        <div
          class="mdc-card mtb-card mdc-card--outlined"
          style="border-radius: 4px;"
        >
          <div
            class="mdc-card__primary-action mtb-card__primary-action mdc-ripple-upgraded"
            tabindex="0"
          >
            <div
              class="mtb-card__media-container"
            >
              <div
                class="mdc-card__media mdc-card__media--16-9 mtb-card__media mtb-card-with-text-under-media"
                style="background-image: url(http://test.loc/test.jpg);"
                tabindex="0"
              />
              <div
                class="mtb-card__media-actions"
              >
                <button
                  aria-label="Remove Card Image"
                  class="components-button mtb-card__media-close-button has-icon"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="dashicon dashicons-no"
                    focusable="false"
                    height="20"
                    role="img"
                    viewBox="0 0 20 20"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M12.12 10l3.53 3.53-2.12 2.12L10 12.12l-3.54 3.54-2.12-2.12L7.88 10 4.34 6.46l2.12-2.12L10 7.88l3.54-3.53 2.12 2.12z"
                    />
                  </svg>
                </button>
              </div>
            </div>
            <div
              class="mtb-card__primary"
            >
              <h2
                aria-label="Title goes here"
                aria-multiline="true"
                class="rich-text block-editor-rich-text__editable mtb-card__title mdc-typography mdc-typography--headline6"
                contenteditable="true"
                role="textbox"
                style="white-space: pre-wrap;"
              >
                Test title
              </h2>
              <h3
                aria-label="Secondary text"
                aria-multiline="true"
                class="rich-text block-editor-rich-text__editable mtb-card__secondary-text mdc-typography mdc-typography--subtitle2"
                contenteditable="true"
                role="textbox"
                style="white-space: pre-wrap;"
              >
                Test secondary text
              </h3>
            </div>
            <div
              aria-label="Supporting text"
              aria-multiline="true"
              class="rich-text block-editor-rich-text__editable mtb-card__supporting-text mtb-card__supporting-text_text-under-media mdc-typography mdc-typography--body2"
              contenteditable="true"
              role="textbox"
              style="white-space: pre-wrap;"
            >
              Test supporting text
            </div>
          </div>
          <div
            class="mdc-card__actions"
          >
            <div
              class="mdc-card__action-buttons"
            >
              <div
                class="mdc-card__action-button-container"
              >
                <button
                  class="mdc-button mdc-card__action mdc-card__action--button"
                >
                  <div
                    aria-label="Button text"
                    aria-multiline="true"
                    class="rich-text block-editor-rich-text__editable"
                    contenteditable="true"
                    role="textbox"
                    style="white-space: pre-wrap;"
                  >
                    Test button action 1
                  </div>
                </button>
              </div>
              <div
                class="mdc-card__action-button-container"
              >
                <button
                  class="mdc-button mdc-card__action mdc-card__action--button"
                >
                  <div
                    aria-label="Button text"
                    aria-multiline="true"
                    class="rich-text block-editor-rich-text__editable"
                    contenteditable="true"
                    role="textbox"
                    style="white-space: pre-wrap;"
                  >
                    Test button action 2
                  </div>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="mdc-card mtb-card mdc-card--outlined"
        style="border-radius: 4px;"
      >
        <div
          class="mdc-card__primary-action mtb-card__primary-action mdc-ripple-upgraded"
          tabindex="0"
        >
          <div
            class="mtb-card__media-container"
          >
            <div
              class="mdc-card__media mdc-card__media--16-9 mtb-card__media mtb-card-with-text-under-media"
              style="background-image: url(http://test.loc/test.jpg);"
              tabindex="0"
            />
            <div
              class="mtb-card__media-actions"
            >
              <button
                aria-label="Remove Card Image"
                class="components-button mtb-card__media-close-button has-icon"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="dashicon dashicons-no"
                  focusable="false"
                  height="20"
                  role="img"
                  viewBox="0 0 20 20"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12.12 10l3.53 3.53-2.12 2.12L10 12.12l-3.54 3.54-2.12-2.12L7.88 10 4.34 6.46l2.12-2.12L10 7.88l3.54-3.53 2.12 2.12z"
                  />
                </svg>
              </button>
            </div>
          </div>
          <div
            class="mtb-card__primary"
          >
            <h2
              aria-label="Title goes here"
              aria-multiline="true"
              class="rich-text block-editor-rich-text__editable mtb-card__title mdc-typography mdc-typography--headline6"
              contenteditable="true"
              role="textbox"
              style="white-space: pre-wrap;"
            >
              Test title
            </h2>
            <h3
              aria-label="Secondary text"
              aria-multiline="true"
              class="rich-text block-editor-rich-text__editable mtb-card__secondary-text mdc-typography mdc-typography--subtitle2"
              contenteditable="true"
              role="textbox"
              style="white-space: pre-wrap;"
            >
              Test secondary text
            </h3>
          </div>
          <div
            aria-label="Supporting text"
            aria-multiline="true"
            class="rich-text block-editor-rich-text__editable mtb-card__supporting-text mtb-card__supporting-text_text-under-media mdc-typography mdc-typography--body2"
            contenteditable="true"
            role="textbox"
            style="white-space: pre-wrap;"
          >
            Test supporting text
          </div>
        </div>
        <div
          class="mdc-card__actions"
        >
          <div
            class="mdc-card__action-buttons"
          >
            <div
              class="mdc-card__action-button-container"
            >
              <button
                class="mdc-button mdc-card__action mdc-card__action--button"
              >
                <div
                  aria-label="Button text"
                  aria-multiline="true"
                  class="rich-text block-editor-rich-text__editable"
                  contenteditable="true"
                  role="textbox"
                  style="white-space: pre-wrap;"
                >
                  Test button action 1
                </div>
              </button>
            </div>
            <div
              class="mdc-card__action-button-container"
            >
              <button
                class="mdc-button mdc-card__action mdc-card__action--button"
              >
                <div
                  aria-label="Button text"
                  aria-multiline="true"
                  class="rich-text block-editor-rich-text__editable"
                  contenteditable="true"
                  role="textbox"
                  style="white-space: pre-wrap;"
                >
                  Test button action 2
                </div>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`blocks: material/card: Edit matches snapshot with a \`horizontal\` card layout. 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      aria-atomic="true"
      aria-live="assertive"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-assertive"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div
      aria-atomic="true"
      aria-live="polite"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-polite"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div>
      <div>
        <div
          class="mdc-card mdc-card--outlined mtb-card mtb-card__list mtb-basic"
          style="border-radius: 4px;"
        >
          <div
            class="mdc-card__primary-action mtb-card__primary-action"
            tabindex="0"
          >
            <div
              class="mtb-card__media-container"
            >
              <div
                class="mdc-card__media mdc-card__media--square mtb-card__media mtb-card-with-text-under-media"
                style="background-image: url(http://test.loc/test.jpg);"
                tabindex="0"
              />
              <div
                class="mtb-card__media-actions"
              >
                <button
                  aria-label="Remove Card Image"
                  class="components-button mtb-card__media-close-button has-icon"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="dashicon dashicons-no"
                    focusable="false"
                    height="20"
                    role="img"
                    viewBox="0 0 20 20"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M12.12 10l3.53 3.53-2.12 2.12L10 12.12l-3.54 3.54-2.12-2.12L7.88 10 4.34 6.46l2.12-2.12L10 7.88l3.54-3.53 2.12 2.12z"
                    />
                  </svg>
                </button>
              </div>
            </div>
            <div
              class="mtb-card__primary"
            >
              <h2
                aria-label="Title goes here"
                aria-multiline="true"
                class="rich-text block-editor-rich-text__editable mtb-card__title mdc-typography mdc-typography--headline6"
                contenteditable="true"
                role="textbox"
                style="white-space: pre-wrap;"
              >
                Test title
              </h2>
              <h3
                aria-label="Secondary text"
                aria-multiline="true"
                class="rich-text block-editor-rich-text__editable mtb-card__secondary-text mdc-typography mdc-typography--subtitle2"
                contenteditable="true"
                role="textbox"
                style="white-space: pre-wrap;"
              >
                Test secondary text
              </h3>
            </div>
          </div>
          <div
            class="mdc-card__actions"
          >
            <div
              class="mdc-card__action-buttons"
            >
              <div
                class="mdc-card__action-button-container"
              >
                <button
                  class="mdc-button mdc-card__action mdc-card__action--button"
                >
                  <div
                    aria-label="Button text"
                    aria-multiline="true"
                    class="rich-text block-editor-rich-text__editable"
                    contenteditable="true"
                    role="textbox"
                    style="white-space: pre-wrap;"
                  >
                    Test button action 1
                  </div>
                </button>
              </div>
              <div
                class="mdc-card__action-button-container"
              >
                <button
                  class="mdc-button mdc-card__action mdc-card__action--button"
                >
                  <div
                    aria-label="Button text"
                    aria-multiline="true"
                    class="rich-text block-editor-rich-text__editable"
                    contenteditable="true"
                    role="textbox"
                    style="white-space: pre-wrap;"
                  >
                    Test button action 2
                  </div>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="mdc-card mdc-card--outlined mtb-card mtb-card__list mtb-basic"
        style="border-radius: 4px;"
      >
        <div
          class="mdc-card__primary-action mtb-card__primary-action"
          tabindex="0"
        >
          <div
            class="mtb-card__media-container"
          >
            <div
              class="mdc-card__media mdc-card__media--square mtb-card__media mtb-card-with-text-under-media"
              style="background-image: url(http://test.loc/test.jpg);"
              tabindex="0"
            />
            <div
              class="mtb-card__media-actions"
            >
              <button
                aria-label="Remove Card Image"
                class="components-button mtb-card__media-close-button has-icon"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="dashicon dashicons-no"
                  focusable="false"
                  height="20"
                  role="img"
                  viewBox="0 0 20 20"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12.12 10l3.53 3.53-2.12 2.12L10 12.12l-3.54 3.54-2.12-2.12L7.88 10 4.34 6.46l2.12-2.12L10 7.88l3.54-3.53 2.12 2.12z"
                  />
                </svg>
              </button>
            </div>
          </div>
          <div
            class="mtb-card__primary"
          >
            <h2
              aria-label="Title goes here"
              aria-multiline="true"
              class="rich-text block-editor-rich-text__editable mtb-card__title mdc-typography mdc-typography--headline6"
              contenteditable="true"
              role="textbox"
              style="white-space: pre-wrap;"
            >
              Test title
            </h2>
            <h3
              aria-label="Secondary text"
              aria-multiline="true"
              class="rich-text block-editor-rich-text__editable mtb-card__secondary-text mdc-typography mdc-typography--subtitle2"
              contenteditable="true"
              role="textbox"
              style="white-space: pre-wrap;"
            >
              Test secondary text
            </h3>
          </div>
        </div>
        <div
          class="mdc-card__actions"
        >
          <div
            class="mdc-card__action-buttons"
          >
            <div
              class="mdc-card__action-button-container"
            >
              <button
                class="mdc-button mdc-card__action mdc-card__action--button"
              >
                <div
                  aria-label="Button text"
                  aria-multiline="true"
                  class="rich-text block-editor-rich-text__editable"
                  contenteditable="true"
                  role="textbox"
                  style="white-space: pre-wrap;"
                >
                  Test button action 1
                </div>
              </button>
            </div>
            <div
              class="mdc-card__action-button-container"
            >
              <button
                class="mdc-button mdc-card__action mdc-card__action--button"
              >
                <div
                  aria-label="Button text"
                  aria-multiline="true"
                  class="rich-text block-editor-rich-text__editable"
                  contenteditable="true"
                  role="textbox"
                  style="white-space: pre-wrap;"
                >
                  Test button action 2
                </div>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
